src/NCNEPortal/Auth/DirectoryService.cs

```
@@ -0,0 +1,72 @@
+        {
+            _secretsConfig = secretsConfig;
+            var redirectUri = isLocalDevelopment ? $"{uriConfig.Value.LocalDevLandingPageHttpsUrl}signin-oidc" :
+                $"{uriConfig.Value.LandingPageUrl}/signin-oidc";
+
+            var authenticationProvider =
+                new MsalAuthenticationProvider(generalConfig.Value.AzureAdClientId,
+                    _secretsConfig.Value.ClientAzureAdSecret,
+                    generalConfig.Value.TenantId,
+                    redirectUri);
+
+            GraphClient = new GraphServiceClient(authenticationProvider, httpProvider);
+        }
+
+        public async Task<IEnumerable<string>> GetGroupMembers()
+        {
+            var groups = new List<Guid> { _secretsConfig.Value.NcGuid, _secretsConfig.Value.NeGuid };
+
+            var users = new List<DirectoryObject>();
+
+            try
+            {
+                foreach (var groupId in groups)
+                {
+                    if (groupId != Guid.Empty)
+                    {
+                        var membersPage = await GraphClient.Groups[groupId.ToString()].Members.Request().GetAsync();
+                        do
+                        {
+                            users.AddRange(membersPage?.CurrentPage);
+
+
+                        } while (membersPage?.NextPageRequest != null &&
+                                 (membersPage = await (membersPage?.NextPageRequest.GetAsync())).Count > 0);
+
+                    }
+                }
+
+                if (users.Count == 0)
+                {
+                    throw new ApplicationException($"Unable to get members of group(s)");
+                }
+
+                return users.Select(o => ((User)o).DisplayName);
+            }
+
+            catch (Exception e)
+            {
+                //TODO: Log!
+                throw new ApplicationException(e.Message);
```

Can we log here?
Can we throw the exception with a message in param 1 with another overload, with the `Exception e` as an inner?

---

Changed the message and passed e as the second parameter